### ISSUE TITLE
Feeds are not displayed in FF71

### ISSUE THREAD

**Issue by voxik** [brief-rss/brief#480]
After update to FF71, Brief lists the number of feeds per channel, but it does not show the feeds themselves. First I thought the DB is corrupted, but after downgrading back to FF70, the feeds are displayed properly, so it must be some FF71 incompatibility. Looking into console, there are errors such as:

~~~
uncaught exception: Object
Error: WebExtension context not found! ExtensionParent.jsm:1229:13
ConstraintError: A mutation operation in the transaction failed because a constraint was not satisfied. ExtensionStorageIDB.jsm:812
    normalizeStorageError resource://gre/modules/ExtensionStorageIDB.jsm:812
    method chrome://extensions/content/child/ext-storage.js:273
    AsyncFunctionThrow self-hosted:695
ConstraintError: A mutation operation in the transaction failed because a constraint was not satisfied. ExtensionStorageIDB.jsm:812
    normalizeStorageError resource://gre/modules/ExtensionStorageIDB.jsm:812
    method chrome://extensions/content/child/ext-storage.js:273
    AsyncFunctionThrow self-hosted:695
IndexedDB UnknownErr: ActorsParent.cpp:613
~~~

~~~
$ rpm -q firefox
firefox-71.0-6.npgo.fc32.x86_64
~~~

**Comment by voxik** [comment]
Just FTR, I have reported it against Fedora FF [[1]], where I was asked to test this with upstream FF binary and it appears to work with upstream release.


[1]: https://bugzilla.redhat.com/show_bug.cgi?id=1779570

**Comment by tanriol** [comment]
You may want to test with the Fedora binary, but still in a new profile. If it works, this may be some kind of profile corruption problem - these do happen in Firefox sometimes, especially on upgrades.

**Comment by tanriol** [comment]
The traceback here is not Brief-specific - this is not the Brief IDB instance, but the one Firefox uses internally for extension's `storage.local` storage area. Brief does use that one as a backup and assumes that is does work without problems.

**Comment by grigi** [comment]
I'm having breakage on FF71 as well, but my console has totally unhelpful messages such as:
```
Brief: opening database in default storage database.js:138:17
Brief: opened database in default storage database.js:171:17
Brief: 17 feeds in database database.js:308:17
Brief: opened database with 34743 entries database.js:92:17
uncaught exception: undefined
Error: An unexpected error occurred 2 ExtensionUtils.jsm
​```
I doubt this is actionable, so I'm just putting it here in case there is a solution. I'm using Brief 2.6.0

**Comment by at46** [comment]
I've the same problem with FF71 on Debian. Browser console shows:
```
IndexedDB UnknownErr: ActorsParent.cpp:613 2
ConstraintError: A mutation operation in the transaction failed because a constraint was not satisfied. 2 ExtensionStorageIDB.jsm:812
```

**Comment by voxik** [comment]
Just FTR, it seems more extensions are affected:

https://bugzilla.redhat.com/show_bug.cgi?id=1779570#c11

**Comment by phorham** [comment]
This issue is fixed on openSUSE release of Firefox 71.0 with this patch:
https://build.opensuse.org/package/rdiff/mozilla/firefox71?linkrev=base&rev=11

**Comment by voxik** [comment]
This seems to be resolved in Fedora, testing with:

~~~
$ rpm -q firefox
firefox-71.0-15.fc32.x86_64
~~~

The upstream fix should be mzbz#1601707. Closing this, since this was FF issue. Thx for your help.
